### PR TITLE
Redis: Apply volatile state results directly to the state model

### LINKED ISSUE #563: Inconsistent volatile state
We only apply certain volatile state values from Redis, which can result in inconsistent view state. All volatile state values from Redis should be applied.

### PR DESCRIPTION
**Pull request by yhabteab**
Since all the inconsistent column/redis key names have been renamed, we don't have to cache redis keys
separately anymore. This PR applies all the `Service/Host` volatile state results directly to the
respective state model.

fixes #563

### CODE REVIEW

**Review by nilmerg — APPROVED** [review]

**Review by lippserd — CHANGES_REQUESTED** [review]
Icinga 2 may not write keys to redis if there is no value. Something like the following is required in DB Web:

```
$data = array_merge(array_fill_keys($columns, null), $data);
```

**Review by nilmerg — CHANGES_REQUESTED** [review]
Please remove class `VolatileState` entirely, it's unused (and obsolete) now.

File: library/Icingadb/Redis/VolatileStateResults.php [hunk 1]
```diff
@@ -114,13 +124,14 @@ protected function applyRedisUpdates()
...
 
-    protected function fetchStates(string $key, array $ids): Generator
+    protected function fetchStates(string $key, array $ids, array $keys): Generator
     {
         $results = IcingaRedis::instance()->getConnection()->hmget($key, $ids);
         foreach ($results as $i => $json) {
             if ($json !== null) {
                 $data = json_decode($json, true);
-                $data = array_intersect_key($data, array_flip(VolatileState::$keys));
+                $data = array_merge(array_fill_keys($keys, null), $data);
```
Comment (nilmerg) [inline library/Icingadb/Redis/VolatileStateResults.php:133]: You're overriding the decoded json data here, leading to a fully unset state.

**Review by nilmerg — CHANGES_REQUESTED** [review]

File: library/Icingadb/Redis/VolatileStateResults.php [hunk 2]
```diff
@@ -114,13 +124,14 @@ protected function applyRedisUpdates()
...
+                $data = array_intersect_key($data, $keys);
```
Comment (nilmerg) [inline library/Icingadb/Redis/VolatileStateResults.php:134]: Where did the array_flip went? Now $data is empty and the redis result is not used??

**Review by nilmerg — CHANGES_REQUESTED** [review]

**Review by nilmerg — APPROVED** [review]

**Pull request merged**